ktlint-rule-engine/src/main/kotlin/com/pinterest/ktlint/rule/engine/internal/KtlintSuppression.kt

```diff
@@ -202,12 +201,15 @@ private fun ASTNode.existingSuppressions() =
     existingSuppressionsFromNamedArgumentOrNull()
         ?: getValueArguments()
 
-private fun ASTNode.existingSuppressionsFromNamedArgumentOrNull() =
-    psi
-        .findDescendantOfType<KtCollectionLiteralExpression>()
-        ?.children
-        ?.map { it.text }
-        ?.toSet()
+private fun ASTNode.existingSuppressionsFromNamedArgumentOrNull(): Set<String>? =
+    recursiveChildren()
+        .firstOrNull { it.elementType == ElementType.COLLECTION_LITERAL_EXPRESSION }
```

Comment:
> Can we leave it without a kdoc since the name is self explanatory? I feel like a kdoc would say "Finds a child with type elementType recursively", which is pretty redundant with the name of the function.
>
> I notice a bunch of other public functions in this file don't have a kdoc.

True, I was aware of the latter. For Ktlint contributors it would suffice, to not document. But those extensions are also meant to be used by external rule providers which might not be familiar with ktlint that much. By providing the API docs, we make it slightly better for them.